src/main/java/com/sintern/service/TestServiceImpl.java

```diff
@@ -0,0 +1,58 @@
+package com.sintern.service;
+
+import com.sintern.domain.dto.TestResultDTO;
+import com.sintern.domain.entity.Student;
+import com.sintern.domain.entity.Test;
+import com.sintern.domain.entity.TestResponse;
+import com.sintern.exception.NoTestsForOpenPositionException;
+import com.sintern.repository.TestRepository;
+import lombok.AccessLevel;
+import lombok.experimental.FieldDefaults;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+@Service
+@FieldDefaults(level = AccessLevel.PRIVATE, makeFinal = true)
+public class TestServiceImpl implements TestService {
+    TestRepository testRepository;
+
+    @Autowired
+    public TestServiceImpl(TestRepository testRepository) {
+        this.testRepository = testRepository;
+    }
+
+    @Override
+    @Transactional
+    public List<TestResultDTO> getTestResultsForOpenPosition(UUID openPositionId) {
+        List<Test> takenTests = testRepository.findAllByOpenInternPositionId(openPositionId);
+        List<TestResultDTO> testResults = new ArrayList<>();
+        if (takenTests.size() == 0) {
+            throw new NoTestsForOpenPositionException("There aren't any test taken for this position yet.");
+        } else {
+            takenTests.forEach(takenTest -> {
```

Comment:
No need for the `else` branch, since we're throwing an **exception** in the `if` block. :smile: